### PR TITLE
Define get/getAll() url option behavior in Windows. [#69]  (Review order: 2)

### LINKED ISSUE #69: Specify the query cookies algorithm.
It's very hard to evaluate how to implement this API without having this algorithm specified.
I can easily guess the meaning of 'name' and 'matchType', but what's the meaning of 'url' when this API is used on window.cookieStore? Is it 'similar' to 'path' or should it be ignored?

### PR DESCRIPTION
**Pull request by inexorabletash**
Make `get()`/`getAll()` reject if `url` option is specified in a Window and it does not match doc's URL.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/cookie-store/pull/71.html" title="Last updated on Jul 26, 2018, 10:20 PM GMT (cef05d5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/cookie-store/71/6a43e82...cef05d5.html" title="Last updated on Jul 26, 2018, 10:20 PM GMT (cef05d5)">Diff</a>

### CODE REVIEW

**Comment by inexorabletash** [comment]
Updated title since this doesn't resolve #69 overall, just the comment.

**Comment by inexorabletash** [comment]
Examples in the explainer have e.g. `{url: '/admin'}`, so we probably need to canonicalize the URL first.

**Comment by inexorabletash** [comment]
Also, URL equality should be done by https://url.spec.whatwg.org/#url-equivalence I suppose.

**Comment by inexorabletash** [comment]
I rebased this on top of #79

File: index.bs [hunk 1]
```diff
@@ -464,11 +513,19 @@ The <dfn method for=CookieStore>getAll(|options|)</dfn> method, when invoked, mu
 
 1. Let |origin| be |environment|’s [=/origin=].
 1. If |origin| is an [=opaque origin=], then return a new promise rejected with a "{{SecurityError}}" {{DOMException}}.
+1. Let |url| be the [=current settings object=]'s [=creation URL=].
+1. If |options|' {{CookieStoreGetOptions/url}} dictionary member is present, then run these steps:
+    1. Let |parsed| be the result of running the [=basic URL parser=] on |options|' {{CookieStoreGetOptions/url}} dictionary member with |url|.
+    1. If the [=current global object=] is a {{Window}} object and |parsed| does not [=url/equal=] |url|,
+        then return a new promise rejected with an "{{InvalidStateError}}" {{DOMException}}.
```
Comment (pwnall) [inline index.bs]: I think we agreed on TypeError for all the errors, in https://github.com/WICG/cookie-store/issues/5

**Pull request merged**